### ISSUE TITLE
ComboBox: use same Nodes for list and button

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/sun#428]
Possibly related to https://github.com/phetsims/sun/issues/405 (instrument ComboBox better).

`ComboBox` initializes its button by creating a (possibly incorrect) `ItemNode`:

```js
    const defaultItemNode = new ItemNode( items[ 0 ], itemWidth, itemHeight, options.itemXMargin, {
      align: options.align
    } );

    // @private button, will be set to correct value when property observer is registered
    this.buttonNode = new ButtonNode( defaultItemNode, _.pick( options, BUTTON_NODE_KEYS ) );
    this.addChild( this.buttonNode );
```

And each time the selection changes, `CombBox` creates a new `ItemNode` for the button:

```js
      this.buttonNode.setItemNode( new ItemNode( item, itemWidth, itemHeight, options.itemXMargin, {
        align: options.align
      } ) );
```

Using scenery's DAG feature, we should be able to use the same set of `ItemNode`s for both the list and the button.  

@samreid @zepumph This would be preferable for PhET-iO too, correct?

**Comment by samreid** [comment]
Yes, if a client customized one entry in the combo box list, they should not have to customize it a parallel but separate way to match the button label.  Enforcing uniformity via the implementation would be great.

**Comment by pixelzoom** [comment]
Note that `ComboBoxItemNode` was renamed to `ItemNode`, see https://github.com/phetsims/sun/issues/429.  I'll update the title/comments in this issue accordingly.

**Comment by pixelzoom** [comment]
@ariel-phet FYI... This should have been a simple change.  But it's gotten hugely complicated because of how a11y was added to ComboBox.   And this is an alarming trend that I'm seeing in all common-code UI components that I've had to modify recently.

**Comment by pixelzoom** [comment]
ComboBox is now using the same ItemNode instances for the list and button.  A consequence of this is that if you change something related to the current selection, it changes in both the list and on the buttons.  For example, Studio screenshots below show the 2 states for the visibility of the "Drink Mix" choice. Note that when visibility is false, the invisible item still highlights in the list and can be selected.   

I'm going to make a note in #405 (instrument Combobox better) that this needs to be revisited as part of PhET-iO design.

![screenshot_956](https://user-images.githubusercontent.com/3046552/50714304-6b3d4780-1035-11e9-9608-2b9634c6fcad.png)

![screenshot_957](https://user-images.githubusercontent.com/3046552/50714308-6e383800-1035-11e9-9ca3-ae0842b65e21.png)

**Comment by pixelzoom** [comment]
Yes, this is the right place for it. 

Those sims are not using ComboBox.createItem to create the combo box items.  They will need to be changed to use ComboBox.createItem, since it now fills in default options.   And in #430, I will be eliminating duck typing of items, and replacing with ComboBox.Item.

**Comment by pixelzoom** [comment]
I verified that failure to call ComboBox.createItem is indeed the issue, by converting twixt in the above commit.    These sims still need to be addressed.  I may not get to this until Monday.

- [ ] masses-and-springs
- [ ] masses-and-springs-basics
- [ ] molecule-shapes
- [ ] molecule-shapes-basics
- [ ] pendulum-lab

**Comment by pixelzoom** [comment]
So the root problem here is that we started with a simple data structure for items.  That data structure was easy to duck type. But then we added additional baggage for PhET-iO and a11y.  This was relatively easy to change in ComboBox.createItem, but sims that were using duck typing had to be modified individually.  And I didn't hit any of those sims when I tested. 

I have converted all of the sims that were not using ComboBox.createItem, documented that items must be created using ComboBox.createItem, and added this assertion to verify that items are wellformed:

```js
assert && assert( item.node instanceof Node && item.value !== undefined && item.options,
  'malformed item - did you create items using ComboBox.createItem?' );
```

@zepumph Thanks for reporting, let me know if you see any other problems.

**Comment by pixelzoom** [comment]
Done in the preceding commits.   ComboBoxItem.node is shared, not ComboBoxListItemNode. 

Closing, will continue further work in #430.
